packages/next-swc/crates/next-custom-transforms/src/transforms/page_static_info/collect_exported_const_visitor.rs

```diff
@@ -0,0 +1,210 @@
+use std::collections::{HashMap, HashSet};
+
+use serde_json::{Map, Number, Value};
+use swc_core::{
+    common::{pass::AstNodePath, Mark, SyntaxContext},
+    ecma::{
+        ast::{
+            BindingIdent, Decl, ExportDecl, Expr, Lit, Pat, Prop, PropName, PropOrSpread, VarDecl,
+            VarDeclKind, VarDeclarator,
+        },
+        utils::{ExprCtx, ExprExt},
+        visit::{AstParentNodeRef, VisitAstPath, VisitWithPath},
+    },
+};
+
+/// The values extracted for the corresponding AST node.
+/// refer extract_expored_const_values for the supported value types.
+/// Undefined / null is treated as None.
+pub enum Const {
+    Value(Value),
+    Unsupported(String),
+}
```

Comment:
I think instead of `Const` enum you should better use `Result<Value, String>`.

Reply:
I'd try it in further refactoring PR.